# zMAP

Isobaric labeling-based mass spectrometry (ILMS) has been widely used to quantify, on a proteome-wide scale, the relative protein abundance in different biological 
conditions. However, large-scale ILMS data sets typically involve multiple runs of mass spectrometry (MS), bringing great computational difficulty to the integration
of ILMS samples. We present zMAP, a toolset that makes ILMS intensities comparable across MS runs by modeling the associated mean-variance dependence and accordingly
applying a variance stabilizing z-transformation. The practical utility of zMAP is demonstrated in several case studies involving the dynamics of cell differentiation
and the heterogeneity across cancer patients.

The design of zMAP aims to simultaneously compare protein profiles of multiple samples and integrate samples from different MS runs for identifying hypervariable 
proteins across samples. zMAP models the mean-variance dependence associated with ILMS intensities and accordingly applies a variance-stabilizing z-transformation
(z-statistic), which dramatically increases the comparability between samples generated by different MS runs. The z-statistic can be widely applied in downstream 
analyses, including PCA, clustering analysis, GSVA, and so on. To facilitate biologists' usage, we offer a user-friendly and straightforward web service. zMAP 
requires that all the involved MS runs are associated with the same biological design, such that the average intensities across all samples in each run are biologically identical.

# Workflow of zMAP
The basic principle of zMAP module is creating a pseudo reference profile separately for each MS run by averaging the included samples.
Then, the M-values of each sample against the corresponding reference profile are scaled based on estimated signal variances, 
producing z-statistics for a final integration across MS runs. Unlike traditional z-score transformation, which is based on observed variance,
zMAP module derives the variance estimates by borrowing strength between proteins with similar intensity levels and fitting a smooth mean-variance curve (MVC) for each MS run. 
Figure 1a depicts the workflow of zMAP module on a single MS run. Technically, after normalizing protein-level ILMS intensities across multiple samples generated by the 
same MS run, zMAP makes a log2 transformation and models the results as following normal distributions, with the mean and variance parameters linked by an unknown MVC.
To fit this MVC, a mean-variance scatter plot for all proteins is drawn, and zMAP uses a sliding window to scan this plot. In this process, the proteins with similar a
verage log2-intensities across samples are grouped together, and a variance estimate is derived for each individual window. For the latter part, zMAP performs a quantile
regression that regresses the observed variances of included proteins against a scaled chi-square distribution by fitting a straight line through the origin, and the slope
is taken as the variance estimate. To avoid the influence of potential hypervariable proteins, which can lead to an overestimated variance, only a certain proportion of 
the proteins with the smallest observed variances are used for the quantile regression. Finally, the MVC is fitted by regressing the variance estimates from all windows
against the corresponding intensity levels. Notably, this sliding-window process equipped with the robust quantile regression can avoid the requirement of biological replicates in each single MS run for fitting an MVC.
![Workflow of zMAP](https://example.com/cat.jpg "zMAP Workflow")

# Try it

A Web-based application of zMAP is provided at http://bioinfo.sibs.ac.cn/shaolab/zMAP. 


# Installation

The latest version release of `motifscanR` could install with:


```r
BiocManager::install('LouisKwok-PICB/motifscanR')
```

# Usage

`motifscanR` could be used for genomic regions motif enrichment analysis
with `motifScan` function. 
```r
library(motifscanR)
library(JASPAR2020)
library(BSgenome.Hsapiens.UCSC.hg19)

# Get a motif pwms
example_motifs <- getJasparMotifs(species = "Homo sapiens",
                                  collection = "CORE")

# Make a set of peaks
peaks <- GenomicRanges::GRanges(seqnames = c("chr1","chr2","chr2"),
                ranges = IRanges::IRanges(start = c(76585873,42772928,100183786),
                                          width = 500))

# Scan motif for example motifs
motif_ix <- motifScan(example_motifs, peaks, genome = "BSgenome.Hsapiens.UCSC.hg19")
```
The input object of genomic regions could be either [GenomicRanges](https://kasperdanielhansen.github.io/genbioconductor/html/GenomicRanges_GRanges.html), [DNAStringSet](https://kasperdanielhansen.github.io/genbioconductor/html/Biostrings.html), 
[DNAString](https://kasperdanielhansen.github.io/genbioconductor/html/Biostrings.html), or character vector. You could see more detail with `?motifScan`

`motifscanR` could also be used for genomic regions motif enrichment analysis
 between two sets of genomic regions with `motifEnrichment` function. 

```r
# Get a motif pwms
example_motifs <- getJasparMotifs(species = "Homo sapiens",
                                  collection = "CORE")
# Make a set of input regions
Input <- GenomicRanges::GRanges(seqnames = c("chr1","chr2","chr2"),
                                ranges = IRanges::IRanges(start = c(76585873,42772928,100183786),
                                                          width = 500))
# Make a set of control regions
Control <- GenomicRanges::GRanges(seqnames = c("chr1","chr3","chr5"),
                                  ranges = IRanges::IRanges(start = c(453123,6524593,100184233),
                                                            width = 500))
# Scan motif for example motifs
motif_ix_input <- motifScan(example_motifs, Input, genome = "BSgenome.Hsapiens.UCSC.hg19")
motif_ix_control <- motifScan(example_motifs, Control, genome = "BSgenome.Hsapiens.UCSC.hg19")

# Find Enrichment motif of input by control
Enrichment_result <- motifEnrichment(motif_ix_input, motif_ix_control)
```
You could type `?motifEnrichment` for more detail.

# Citation

To cite the `zMAP` package in publications, please use


